docs/docs/dev_docs/tutorials/token_portal/setup.md

```diff
@@ -0,0 +1,278 @@
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "strict": true,
+    "declaration": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "downlevelIteration": true,
+    "inlineSourceMap": true,
+    "declarationMap": true,
+    "importHelpers": true,
+    "resolveJsonModule": true,
+    "composite": true,
+    "skipLibCheck": true
+  },
+  "include": [
+    "packages/src/**/*",
+    "contracts/**/*.json",
+    "packages/src/**/*",
+    "packages/aztec-contracts/token_bridge/target/*.json"
+  ],
+  "exclude": ["node_modules", "**/*.spec.ts", "contracts/**/*.ts"],
+  "references": []
+}
+```
+
+# Create a nargo project
+
+Now inside `aztec-token-bridge` create a new directory called `aztec-contracts`
+
+Inside `aztec-contracts`, create a nargo contract project by running
+
+```bash
+mkdir aztec-contracts
+cd aztec-contracts
+nargo new -contract token_bridge
+```
+
+Your structure will look something like this
+
+```
+aztec-contracts
+└── token_bridge
+    ├── Nargo.toml
+    ├── src
+		  ├── main.nr
+```
+
+Inside `Nargo.toml` you will need to add some dependencies. Put this at the bottom:
+
+```json
+[dependencies]
+aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="#include_aztec_version", directory="yarn-project/aztec-nr/aztec" }
+value_note = { git="https://github.com/AztecProtocol/aztec-packages/", tag="#include_aztec_version", directory="yarn-project/aztec-nr/value-note"}
+safe_math = { git="https://github.com/AztecProtocol/aztec-packages/", tag="#include_aztec_version", directory="yarn-project/aztec-nr/safe-math"}
+```
+
+Inside `src` you will see a `main.nr` file. This is where our main smart contract will go.
+
+We will also be writing some helper functions that should exist elsewhere so we don't overcomplicated our contract. In `src` create another file called `util.nr` so your dir structure should now look like this:
+
+```
+aztec-contracts
+└── token_bridge
+    ├── Nargo.toml
+    ├── src
+      ├── main.nr
+      ├── util.nr
+```
+
+# Create a hardhat project
+
+In the root dir `aztec-token-bridge`, create a new directory called `l1-contracts` and run `npx hardhat init` inside of it. Keep hitting enter so you get the default setup.
+
+```bash
+mkdir l1-contracts
+cd l1-contracts
+npx hardhat init
+```
+
+Once you have a hardhat project set up, delete the `contracts` directory inside `l1-contracts`. We will be cloning a new `contracts` dir in the next step.
+
+```bash
+rm -rf contracts
+```
+
+## Download Ethereum contracts
+
+We will write the `TokenPortal.sol` contract in this tutorial, but it has many imports that we will need to have locally.
+
+To make this easier we have a standalone submodule on GItHub with all the smart contracts with relative paths - [find it in the devrels repo under utils](https://github.com/AztecProtocol/dev-rel/tree/main/utils). You can clone this directly into `l1-contracts` (recommended to then `rm -rf .git`).
```

Comment:
Should we link directly to this [repo](https://github.com/catmcgee/contracts/tree/main) instead of the dev rel repo that links to it?

Reply:
```suggestion
To make this easier we have a standalone submodule on GItHub with all the smart contracts with relative paths - [find it in the devrels repo under utils](https://github.com/AztecProtocol/dev-rel/tree/main/utils). You can clone this directly into `l1-contracts` (recommended to then `rm -rf .git` in the `contracts` dir).
```